### PR TITLE
new gromacs tool for calculating RMSF using gmx

### PR DESCRIPTION
**Pull request by thepineapplepirate**

### CODE REVIEW

File: tools/gromacs/rmsf.xml [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+            <option value="none">Raw Data.</option>
+        </param>
+        <expand macro="log"/>
+    </inputs>
+    <outputs>
+        <data name="output1" format="xvg" from_work_dir="rmsf.xvg" label="GROMACS calculation of RMSF on ${on_string}"/>
+        <expand macro="log_outputs"/>
+    </outputs>
+    <tests>
+        <test>
```
Comment (bgruening) [inline tools/gromacs/rmsf.xml:52]: expect_num_outputs

File: tools/gromacs/rmsf.xml [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        <param name="ndx_input" type="data" format="ndx" label="Index (NDX) file" help="Index file containing your system's different atomic and molecular groups."/>
+        <param name="index" value="0" min="0" type="integer" label="Index of group" help="Index of group for calculating the Root Mean Square Fluctuation (RMSF) - i.e. the group's position in the ndx file (using zero-based numbering, so the first group has index 0)."/>
+        <param name="resavg" type="select" label="Calculate average RMSF for entire residues?" help="This option works best if the selected index group mainly includes whole residues.">
+            <option value="false">No</option>
+            <option value="true">Yes</option>
+        </param>
+        <param name="outputformat" type="select" label="Output format">
+            <option value="xmgrace">Xmgrace. This is the default way.</option>
+            <option value="none">Raw Data.</option>
```
Comment (bgruening) [inline tools/gromacs/rmsf.xml:43]: remove dots at the end

File: tools/gromacs/rmsf.xml [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        <import>macros.xml</import>
+        <token name="@GALAXY_VERSION@">0</token>
+    </macros>
+    <expand macro="requirements"/>
+    <command detect_errors="exit_code"><![CDATA[
+
+    ln -s '$traj_input' ./traj_input.${traj_input.ext} &&
+    ln -s '$structure_input' ./structure_input.${structure_input.ext} &&
+    #if '$ndx_input':
+    ln -s '$ndx_input' ./ndx_input.${ndx_input.ext} &&
```
Comment (bgruening) [inline tools/gromacs/rmsf.xml:13]: indentation

Comment (thepineapplepirate) [reply tools/gromacs/rmsf.xml:13]: So this was automatically indented (or not indented) this way, when i formatted the file using the galaxy extension, but maybe it missed this. will correct.

File: tools/gromacs/rmsf.xml (hunk 2 above)
Comment (thepineapplepirate) [reply tools/gromacs/rmsf.xml]: removed

File: tools/gromacs/rmsf.xml (hunk 3 above)
Comment (thepineapplepirate) [reply tools/gromacs/rmsf.xml]: indented

**Comment by simonbray** [comment]
Sorry, won't have time to finish reviewing this and the other PR this evening. But it looks good to me. :+1: 

Just one thing that struck me: is the 'raw data' option really that useful? Would it be worth using the `xvg_or_tab_outputs` macro so the user can choose to download a Galaxy tabular file as well?

**Comment by thepineapplepirate** [comment]
> Sorry, won't have time to finish reviewing this and the other PR this evening. But it looks good to me. 👍
> 
> Just one thing that struck me: is the 'raw data' option really that useful? Would it be worth using the `xvg_or_tab_outputs` macro so the user can choose to download a Galaxy tabular file as well?

I think it's useful, depending on how a user chooses to plot the data, in my experience its been useful to just have the raw data points.

**Review by bgruening — APPROVED** [review]

**Pull request merged**